packages/idyll-cli/src/pipeline/index.js

```diff
@@ -30,115 +30,107 @@ const build = (opts, paths, resolvers) => {
-          (acc, { name, source, async }) => {
-            let { resolvedName, data } = resolvers
-              .get('data')
-              .resolve(name, source, async);
-            acc[resolvedName] = data;
-            return acc;
-          },
-          {}
-        );
-
-        const css = resolvers.get('css').resolve();
-
-        output = {
-          ast: ast,
-          components,
-          data,
-          css,
-          syntaxHighlighting: getHighlightJS(ast, paths),
-          context: opts.context,
-          opts: {
-            ssr: opts.ssr,
-            theme: opts.theme,
-            layout: opts.layout,
-            authorView: opts.authorView
-          }
-        };
-        if (!opts.ssr) {
-          output.html = getBaseHTML(ast, template, opts);
-        } else {
-          output.html = getHTML(
-            paths,
-            ast,
-            output.components,
-            output.data,
-            template,
-            opts
-          );
+  if (!opts.ssr) {
+    output.html = getBaseHTML(ast, template, opts);
+  } else {
+    output.html = getHTML(
+      paths,
+      ast,
+      output.components,
+      output.data,
+      template,
+      opts
+    );
+  }
+
+  let js = await bundleJS(opts, paths, output); // create index.js bundle
+
+  // minify bundle if necessary and store it
+  if (opts.minify) {
+    js = Terser.minify(js, {
+      mangle: { keep_fnames: true }
+    }).code;
+  }
+  output.js = js;
+
+  // write output
+  await Promise.all([
+    writeFile(paths.JS_OUTPUT_FILE, output.js),
+    writeFile(paths.CSS_OUTPUT_FILE, output.css),
+    writeFile(paths.HTML_OUTPUT_FILE, output.html),
+    pathExists(paths.STATIC_DIR)
+      .then(exists => {
+        if (exists && paths.STATIC_DIR !== paths.STATIC_OUTPUT_DIR) {
+          return copy(paths.STATIC_DIR, paths.STATIC_OUTPUT_DIR);
         }
-      });
-    })
-    .then(() => {
-      return bundleJS(opts, paths, output); // create index.js bundle
-    })
-    .then(js => {
-      // minify bundle if necessary and store it
-      if (opts.minify) {
-        js = Terser.minify(js, {
-          mangle: { keep_fnames: true }
-        }).code;
-      }
-      output.js = js;
-    })
-    .then(() => {
-      return Promise.all([
-        writeFile(paths.JS_OUTPUT_FILE, output.js),
-        writeFile(paths.CSS_OUTPUT_FILE, output.css),
-        writeFile(paths.HTML_OUTPUT_FILE, output.html),
-        pathExists(paths.STATIC_DIR)
-          .then(exists => {
-            if (exists && paths.STATIC_DIR !== paths.STATIC_OUTPUT_DIR) {
-              return copy(paths.STATIC_DIR, paths.STATIC_OUTPUT_DIR);
-            }
-            return null;
-          })
-          .catch(e => {
-            debug('Error copying static files', e);
-          })
-      ]);
-    })
-    .then(() => {
-      return output; // return all results
-    });
+        return null;
+      })
+      .catch(e => {
+        debug('Error copying static files', e);
+      })
```

Comment:
I think we may want to just let this error be thrown, so it can be handled at the top level by https://github.com/idyll-lang/idyll/blob/jh/ast-option/packages/idyll-cli/src/index.js#L283-L297

Reply:
Although I don't have super strong feelings about this if you think handling the errors locally is a better approach.

Reply:
Up to you! I didn't add this, it just carries over from your prior implementation. Things are just rearranged because I moved the code from a `Promise.then` chain to a "flatter" series of `await` calls.

Reply:
Ah got it! Then let's keep this PR as it is, that's a separate issue.